compiler/rustc_hir/src/hir.rs

```diff
@@ -305,12 +305,12 @@ pub struct GenericArgs<'hir> {
     pub parenthesized: bool,
 }
 
-impl<'tcx> GenericArgs<'tcx> {
+impl GenericArgs<'_> {
     pub const fn none() -> Self {
         Self { args: &[], bindings: &[], parenthesized: false }
     }
 
-    pub fn inputs(&self) -> &[Ty<'tcx>] {
+    pub fn inputs(&self) -> &[Ty<'_>] {
```

Comment:
This change looks incorrect. `'_` will turn into the lifetime of `&self` by the elision rules, not `'tcx`. That couled be an issue if `GenericArgs` is dropped but you still want to use the types afterward.

Reply:
This is how it is on master: https://github.com/rust-lang/rust/blob/b32e6e6ac8921035177256ab6806e6ab0d4b9b94/compiler/rustc_hir/src/hir.rs#L313

You are right though. In fact, that was the original motivation for why I *was* going to PR https://github.com/rust-lang/rust/compare/master...jackh726:binder-refactor-part-1 before I realized that I don't need those changes for this PR anymore. Ultimately, this is just a revert to try to minimize the diff. (Commits for this branch are messy at the moment, but I'm splitting things into separate PRs as I can, so much of these will reduced/removed anyways.